src/query/v2/plan/scoped_profile.hpp

```diff
@@ -13,13 +13,47 @@
 
 #include <cstdint>
 
+#include <json/json.hpp>
+
 #include "query/v2/context.hpp"
 #include "query/v2/plan/profile.hpp"
 #include "utils/likely.hpp"
 #include "utils/tsc.hpp"
 
 namespace memgraph::query::v2::plan {
 
+class ScopedCustomProfile {
```

Comment:
Could you elaborate why we need `ScopedCustomProfile`?

Reply:
My main goal was to provide an abstraction that is similar for `ScopedProfile` and how it is used while it is a bit more flexible in terms of what we can measure. 

That's why I came up with the idea of `ScopedCustomProfile`: it provides the same way of measuring thing, but as the results are stored in a JSON object, it can be extended without changing how they are communicated to the user.

Reply:
Ahaa I understand your intention, it is ok for me